### PR TITLE
ocamltest should report skipped test as skipped rather than passed

### PR DESCRIPTION
**Pull request by shindere**
Proposed fix for issue #13481 reported earlier today.

### CODE REVIEW

File: ocamltest/main.ml [hunk 1]
```diff
@@ -95,6 +95,7 @@ let join_result summary result =
   | Fail, _
   | _, Some_failure -> Some_failure
   | Skip, All_skipped -> All_skipped
+  | Pass, All_skipped -> All_skipped
```
Comment (nojb) [inline ocamltest/main.ml]:
```suggestion
  | (Skip | Pass), All_skipped -> All_skipped
```

**Comment by shindere** [comment]
Implemented, thanks.

**Review by nojb — APPROVED** [review]
LGTM

**Comment by shindere** [comment]
Nicolás Ojeda Bär (2024/09/26 06:25 -0700):
> @nojb approved this pull request.
>
> LGTM

Cool, thanks a lot!

**Comment by gasche** [comment]
The present fix was incorrect (as discussed in #13484), it now reports passing tests as skipped. I reverted it from trunk.

**Comment by nojb** [comment]
> The present fix was incorrect (as discussed in #13484), it now reports passing tests as skipped. I reverted it from trunk.

Thanks. I had an inkling that something was a bit off, but summarily disregarded it out of lazyness. I should not review PRs when ill :)

**Pull request merged**